### PR TITLE
Add per-domain options (and GitHub token)

### LINKED ISSUE #114: Issue with the merge commit tweaks
From: https://github.com/chalk/ansi-styles/commits/master

screen shot 2016-03-30 at 14 05 50

### LINKED ISSUE #1714: RGH handles only one API Token
Currently if you use RGH on a different domain and GitHub, as soon as you provide a token, the other site no longer works (as expected).
```
features.js:79 RefinedGitHubAPIError: The token seems to be incorrect or expired. Update it in the options.
    at call (chrome-extension://mnoacfgkaplmhhcgdnekddifjfcllloj/content.js:2481:11)
    at <anonymous>
```
We need to add the option to specify what URL the token is for and only use it for that domain. This becomes more important if #1713 is merged.

### PR DESCRIPTION
**Pull request by fregante**
Closes #1714 
Closes #114 
Closes https://github.com/sindresorhus/module-requests/issues/114
Replaces #1384

With [recent changes to `webext-options-sync`](https://github.com/fregante/webext-options-sync/compare/v0.21.2...master) and the slew of [new webext modules](https://github.com/fregante?utf8=%E2%9C%93&tab=repositories&q=webext&type=source&language=) it's now possible to:

- [Detect new domain permissions](https://github.com/fregante/webext-additional-permissions) (when you "Enable RGH on this domain")
- Automatically show the domain selector _if_ necessary
- Switch between domains in the options

In short, you can now have per-domain GitHub tokens and _all_ other settings, with relatively little effort in Refined GitHub (refer to `options-storage.ts` and `options.tsx` only)

# Testing

This seems to work but testing and code quality is limited. Please review.

cc Enterprise users @simonkotwicz @gdaunton @ahatzz11 @mrchief @ahusseini @snappieT @kj800x @rastersize @tknickman @gibfahn @rastersize @SivanMehta @NickCraver (please react with 👀 if you don't want to be pinged for future GHE updates)

# Domain picker demo

### CODE REVIEW

**Comment by fregante** [comment]
This was tested in Chrome and Firefox and it looks like it's working on both custom domains and existing github.com options :tada:

I'm leaving this open for a couple more days just to allow some reviews

**Comment by fregante** [comment]
Code reviews are still welcome but I wanted to get this in and avoid further big conflicts

**Comment by rastersize** [comment]
@fregante I had some time to restart my browser (Brave, version 0.67.123 Chromium: 76.0.3809.87) so the extension updated. However whenever I switch between GitHub.com and my GHE domain from the dropdown the API token stays the same. I.e. it only shows one of them. I’m not sure if it uses the same token for both or not though.

**Comment by rastersize** [comment]
Just tried with Chrome (v76) as well and the same thing happened there.
